server/attr/dictionaries.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attr
+
+// The code in here is pretty simple. It has a global lock that's held whenever doing anything.
+// So this isn't great for high contention scenarios. The code also iterates through the dictionaries
+// while holding the lock.
+//
+// This is all fine and dandy, since this code ends up being seldom used. Typically, this
+// is called only when a gRPC stream is initiated, which is a pretty rare situation.
+// And typically, there will be 1 dictionary for all of the mixer.
+
+import (
+	"sync"
+)
+
+type dictionary map[int32]string
```

Comment:
Is this the vocabulary of all names?
Is the assertion that we will have multiple vocabularies in use at any given time, and this allows for efficient sharing between them?